Core/Src/bmb.c

```diff
@@ -133,100 +137,147 @@ void initBmbs(uint32_t numBmbs)
+					// Read brick voltage in [15:2]
+					uint32_t brickVRaw = ((recvBuffer[4 + 2*j] << 8) | recvBuffer[3 + 2*j]) >> 2;
+					float brickV = brickVRaw * CONVERT_14BIT_TO_5V;
+					bmb[j].brickV[i] = brickV;
+				}
+			}
+			else
+			{
+				printf("Error during cellReg readAll!\r\n");
+
+				// Failed to acquire data. Set status to MIA
+				for (int j = 0; j < numBmbs; j++)
+				{
+					bmb[j].brickVStatus[i] = MIA;
+				}
+			}
+		}
+
+		// Read VBLOCK register
+		if (readAll(VBLOCK, recvBuffer, numBmbs))
+		{
+			for (uint8_t j = 0; j < numBmbs; j++)
+			{
+				// Read block voltage in [15:2]
+				uint32_t blockVRaw = ((recvBuffer[4 + 2*j] << 8) | recvBuffer[3 + 2*j]) >> 2;
+				float blockV = blockVRaw * CONVERT_14BIT_TO_60V;
+				bmb[j].blockV = blockV;
+			}
+		}
+		else
+		{
+			printf("Error during VBLOCK readAll!\r\n");
+
+			// Failed to acquire data. Set status to MIA
+			for (int j = 0; j < numBmbs; j++)
+			{
+				bmb[j].blockVStatus = MIA;
+			}
+		}
+
+		// TODO Add check - Compare VBLOCK with sum of brick voltages
+
+		// Read AUX/TEMP registers
 		for (int auxChannel = AIN1; auxChannel <= AIN2; auxChannel++)
 		{
-			// Temperature index for Bmb_S struct board temp
-			int tempIdx = muxGpio - MUX7 + ((auxChannel == AIN2) ? 2 : 0);
-			// Read temperature from BMBs
 			if (readAll(auxChannel, recvBuffer, numBmbs))
 			{
 				// Parse received data
-				for (uint8_t i = 0; i < numBmbs; i++)
+				for (uint8_t j = 0; j < numBmbs; j++)
 				{
 					// Read AUX voltage in [15:4]
-					uint32_t auxRaw = ((recvBuffer[4 + 2*i] << 8) | recvBuffer[3 + 2*i]) >> 4;
+					uint32_t auxRaw = ((recvBuffer[4 + 2*j] << 8) | recvBuffer[3 + 2*j]) >> 4;
 					float auxV = auxRaw * CONVERT_12BIT_TO_3V3;
-					// Determine boardTempVoltage index for current reading
-					bmb[i].boardTempVoltage[tempIdx] = auxV;
+					bmb[j].tempVoltage[muxState + ((auxChannel == AIN2) ? NUM_MUX_CHANNELS : 0)] = auxV;
+
+					// Convert temp voltage registers to temperature readings
+					if(muxState == (MUX7 || MUX8)) // NTC/ON-Board Temp Channel
+					{
+						bmb[j].boardTemp[muxState - MUX7 + ((auxChannel == AIN2) ? 2 : 0)] = lookup(auxV, &ntcTable);
+					}
+					else // Zener/Brick Temp Channel
+					{
+						bmb[j].brickTemp[muxState + ((auxChannel == AIN2) ? NUM_MUX_CHANNELS : 0)] = lookup(auxV, &zenerTable);
+					}
 				}
 			}
 			else
 			{
 				// Failed to acquire data. Set status to MIA
-				for (int i = 0; i < numBmbs; i++)
+				for (int j = 0; j < numBmbs; j++)
 				{
-					bmb[i].boardTempStatus[tempIdx] = MIA;
+					bmb[j].tempStatus[muxState + ((auxChannel == AIN2) ? 8 : 0)] = MIA;
 				}
 			}
 		}
-	}
 
-	for (uint8_t i = 0; i < numBmbs; i++)
-	{
-		for (int j = 0; j < NUM_BOARD_TEMP_PER_BMB; j++)
-		{
-			float ntcV = bmb[i].boardTempVoltage[j];
+		// Cycle to next MUX configuration
+		setMux(numBmbs, (muxState + 1) % NUM_MUX_CHANNELS	);
 
-			bmb[i].boardTemp[j] = lookup(ntcV, &ntcTable);
-		}
+		// Update lastUpdate
+		lastUpdate = HAL_GetTick();
```

Comment:
How long do you think it takes to run through the entire function code? Because if it's ~1ms then the time difference will actually be more than 100ms and slowly drift out. Not a huge deal but you might want to move this towards the beginning of the function